packages/main/src/Input.ts

```diff
@@ -1235,9 +1232,8 @@ class Input extends UI5Element implements SuggestionComponent, IFormInputElement
 	 * Updates the input value on item select.
 	 * @param item The item that is on select
 	 */
-	updateValueOnSelect(item: SuggestionListItem) {
-		const nonSelectable = item.type === "Inactive" || item.groupItem;
-		const itemValue = nonSelectable ? this.valueBeforeSelectionStart : (item.effectiveTitle || item.textContent || "");
+	updateValueOnSelect(item: IInputSuggestionItem) {
+		const itemValue = item.hasAttribute("ui5-suggestion-item-group") ? this.valueBeforeSelectionStart : (item as IInputSuggestionItemSelectable).text;
```

Comment:
Maybe extract the check for group attribute in a method for better readability. I noticed that it is used in several places.